src/stream_type.rs

```diff
@@ -52,4 +54,38 @@ impl io::Write for StreamType {
     }
 }
 
-impl Io for StreamType {}
+impl AsyncRead for StreamType {
+    unsafe fn prepare_uninitialized_buffer(&self, buf: &mut [u8]) -> bool {
```

Comment:
Is there a measurable performance gain that this provides? I read the docs about the functions you're delegating to here and it looks fine. In general though it seems good to justify usage of unsafe in a comment, preferably with perf numbers.